keypresses lost/eaten occasionally

If I open bemenu-run and type stuff, I lose two letters. I don't know the exact interval, and it's not always the same letters. I tried to type "the quick brown fox jumps over the lazy dog" two times. The first time I lost the "wn" in "brown". The second time I think I lost part of the word "jumps" so it said "jus". It's the sorta thing that could drive a man crazy.
I'm on version 0.3.0 from the NixOS repos, so apologies if this bug has already been fixed in a later version. I searched the issues for "keypress" and saw nothing.
I believe this happens with bemenu in general, by the way. I also have a clipboard script that uses it and seems to lose keypresses, so it's not specific to the run command.
I'm using Sway (wayland), if that matters.

I've never built (unpackaged) things from source on NixOS and it's looking to be a pain, so I might not be able to figure this out any time soon. I cloned the repo, made the changes in that file, and ran `make`, but then was greeted with some errors.
```
cc -g -O2 -Wall -Wextra -Wpedantic -Wformat=2 -Wstrict-aliasing=3 -Wstrict-overflow=5 -Wstack-usage=12500 -Wfloat-equal -Wcast-align -Wpointer-arith -Wchar-subscripts -Warray-bounds=2 -Wno-unknown-warning-option -std=c99 -D_FORTIFY_SOURCE=2 -DBM_VERSION=\"0.4.0\" -DBM_PLUGIN_VERSION=\"0.4.0-1873f7460fe9e99c405f99f97f82ae67f783b48f\" -DINSTALL_LIBDIR=\"/usr/local/lib\" -D_DEFAULT_SOURCE -Ilib   -c lib/3rdparty/cdl.c  -o cdl.a
make: cc: Command not found
make: *** [GNUmakefile:36: cdl.a] Error 127

--- a/lib/renderers/wayland/registry.c
+++ b/lib/renderers/wayland/registry.c
@@ -107,8 +107,15 @@ keyboard_handle_leave(void *data, struct wl_keyboard *keyboard, uint32_t serial,
 static void
 press(struct input *input, xkb_keysym_t sym, uint32_t key, enum wl_keyboard_key_state state)
 {
-    input->sym = (state == WL_KEYBOARD_KEY_STATE_PRESSED ? sym : XKB_KEY_NoSymbol);
-    input->code = (state == WL_KEYBOARD_KEY_STATE_PRESSED ? key + 8 : 0);
+
+    if (state == WL_KEYBOARD_KEY_STATE_PRESSED) {
+	    input->sym = sym;
+	    input->code = key + 8;
+	    input->key_pending = true;
+    } else if (!input->key_pending) {
+	    input->sym = XKB_KEY_NoSymbol;
+	    input->code = 0;
+    }
 
     if (input->notify.key)
         input->notify.key(state, sym, key);

--- a/lib/renderers/wayland/wayland.c
+++ b/lib/renderers/wayland/wayland.c
@@ -72,6 +72,10 @@ poll_key(const struct bm_menu *menu, unsigned int *unicode)
     wayland->input.sym = XKB_KEY_NoSymbol;
     wayland->input.code = 0;
 
+    if (!wayland->input.key_pending)
+	    return BM_KEY_UNICODE;
+    wayland->input.key_pending = false;
+
     switch (sym) {
         case XKB_KEY_Up:
             return BM_KEY_UP;
@@ -381,6 +385,7 @@ constructor(struct bm_menu *menu)
     wayland->fds.display = wl_display_get_fd(wayland->display);
     wayland->fds.repeat = timerfd_create(CLOCK_MONOTONIC, TFD_CLOEXEC | TFD_NONBLOCK);
     wayland->input.repeat_fd = &wayland->fds.repeat;
+    wayland->input.key_pending = false;
     recreate_windows(menu, wayland);
 
     if (!efd && (efd = epoll_create1(EPOLL_CLOEXEC)) < 0)

--- a/lib/renderers/wayland/wayland.h
+++ b/lib/renderers/wayland/wayland.h
@@ -65,6 +65,8 @@ struct input {
     struct {
         void (*key)(enum wl_keyboard_key_state state, xkb_keysym_t sym, uint32_t code);
     } notify;
+
+    bool key_pending;
 };
 
 struct buffer {

